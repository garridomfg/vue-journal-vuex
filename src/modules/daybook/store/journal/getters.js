
/* export const myGetter = ( state ) => {
    return state.something
} */

export const getEntriesByTerm = ( /* state */ ) => {
    
}

export const getEntryById = ( /* state */ ) => {
    
}