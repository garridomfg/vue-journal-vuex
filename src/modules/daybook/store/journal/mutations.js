
/* export const myMutation = ( state ) => {
    
} */

export const setEntries = (state, entries) => {
    state.entries = [...state.entries, ...entries]
    state.isLoading = false
}

export const updateEntry = (state, entry) => {
    const index = state.entries.map(element => element.id).indexOf(entry.id)
    state.entries[index] = entry
}

export const addEntry = (state, entry) => {
    state.entries = [entry, ...state.entries]
}

export const deleteEntry = (state, id) => {
    state.entries = state.entries.filter(entry => entry.id !== id)
}

export const clearEntries = (state) => {
    state.entries = []
}