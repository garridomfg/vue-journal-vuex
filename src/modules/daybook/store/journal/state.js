
export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Esse commodo voluptate ullamco enim esse nisi nulla pariatur labore qui do elit cillum deserunt. Culpa sunt excepteur eu voluptate. Fugiat sunt est enim do ad anim mollit anim ullamco incididunt. Non aliquip laboris culpa laborum exercitation cupidatat ut aliqua ad veniam est occaecat officia.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'Quis magna qui elit deserunt id eiusmod consectetur voluptate incididunt. Sint laborum non aliqua veniam. Nostrud velit laborum sit proident ea sit cillum laboris aliqua exercitation minim officia. Nisi ipsum et laboris cillum excepteur fugiat. Pariatur velit labore consequat ut aliqua officia labore incididunt aliquip. Adipisicing non sint sit veniam in enim. Ut excepteur adipisicing sit ad officia nisi ea excepteur.',
            picture: null,
        },
        {
            id: new Date().getTime() + 2000,
            date: new Date().toDateString(),
            text: 'In Lorem laborum commodo sint cillum aliqua fugiat. Laboris eu eiusmod laboris aliquip tempor consectetur id eu sit aliqua in. Laboris Lorem proident pariatur sunt consequat est minim quis nostrud dolor velit elit velit.',
            picture: null,
        },
    ],
})