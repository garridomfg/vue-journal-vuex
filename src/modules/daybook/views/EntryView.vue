<template>
  <div class="entry-title d-flex justify-content-between p-2">
    <div>
      <span class="text-success fs-3 fw-bold">15</span>
      <span class="mx-1 fs-3">Julio</span>
      <span class="mx-2 fs-4 fw-light">2021, jueves</span>
    </div>

    <div>
      <button class="btn btn-danger mx-2">
        Delete
        <i class="fa fa-trash-alt"></i>
      </button>
      <button class="btn btn-primary">
        Delete
        <i class="fa fa-upload"></i>
      </button>
    </div>
  </div>

  <hr />
  <div class="d-flex flex-column px-3 h-75">
    <textarea placeholder="¿What´s happened today?"></textarea>
  </div>

  <Fab icon="fa-save" />

  <img
    class="img-thumbnail"
    src="https://calzadoslola.com/wp-content/uploads/2020/07/travel-landscape-01.jpg"
    alt="entry-picture"
  />
</template>

<script>
import { defineAsyncComponent } from "vue";
export default {
  components: {
    Fab: defineAsyncComponent(() => import("../components/Fab.vue")),
  },
};
</script>

<style lang="scss" scoped>
textarea {
  border: none;
  font-size: 20px;
  height: 100%;

  &:focus {
    outline: none;
  }
}
img {
  bottom: 150px;
  box-shadow: 0px 5px 10px rgba($color: #000000, $alpha: 0.2);
  position: fixed;
  right: 20px;
  width: 200px;
}
</style>
