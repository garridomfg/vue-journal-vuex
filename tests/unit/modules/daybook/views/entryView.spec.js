import { createStore } from 'vuex';
import { shallowMount } from '@vue/test-utils'
import journal from '@/modules/daybook/store/journal'
import Swal from 'sweetalert2'
import EntryView from '@/modules/daybook/views/EntryView'
import { journalState } from '../../../mock-data/test-journal-state'

const createVuexStore = (initialState) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))

describe('EntryView', () => {
    const store = createVuexStore(journalState)
    store.dispatch = jest.fn()

    const mockRouter = {
        push: jest.fn()
    }

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()

        wrapper = shallowMount(EntryView, {
            props: {
                id: '-MiceyoT-TsPbU9NRtjC'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
    })

    test('should reject the user because the id doesnt exist ', () => {
        const wrapper = shallowMount(EntryView, {
            props: {
                id: 'This ID doesn´t exist in Store'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store],
            }
        })

        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    });

    test('should show the entry correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalled()
    });

    test('should delete the entry and redirect', (done) => {

        Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))

        wrapper.find('.btn-danger').trigger('click')

        expect(Swal.fire).toHaveBeenCalledWith({
            title: "¿Are you sure?",
            text: "Once deleted it cannot be recovered",
            showDenyButton: true,
            confirmButtonText: "Yes, i´m sure",
        })

        setTimeout(() => {
            expect(store.dispatch).toHaveBeenLastCalledWith("journal/deleteEntry", "-MiceyoT-TsPbU9NRtjC")
            expect(mockRouter.push).toHaveBeenCalled()
            done()
        }, 1);
    });
});