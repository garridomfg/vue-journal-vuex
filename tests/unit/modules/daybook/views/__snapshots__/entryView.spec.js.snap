// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryView should show the entry correctly 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">2</span><span class="mx-1 fs-3">September</span><span class="mx-2 fs-4 fw-light">2021, Thursday</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Delete <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Upload picture <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿What´s happened today?"></textarea></div>
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
